crates/plugin_tree_shake/src/module.rs

```diff
@@ -221,458 +224,134 @@ impl TreeShakeModule {
       used_idents.insert(used_ident);
     }
 
-    {
-      farmfe_core::farm_profile_scope!(format!(
-        "analyze self executed stmts {:?}",
-        self.module_id.to_string()
-      ));
-      let unresolved_mark = Mark::from_u32(module.meta.as_script().unresolved_mark);
-      let (ids, mut items_map) = ModuleAnalyze::analyze(module);
-      let mut module_analyze = ModuleAnalyze::new();
-      let mut entries: HashSet<ItemId> = HashSet::new();
-
-      // remove unused export specifiers
-      for (stmt, used_idents) in stmt_used_idents_map.iter() {
-        let default_str = UsedIdent::Default.to_string();
-        let used_idents_string = used_idents
-          .iter()
-          .filter_map(|ident| match ident {
-            UsedIdent::SwcIdent(ident) => Some(ident),
-            UsedIdent::Default => Some(&default_str),
-            _ => None,
-          })
-          .collect::<HashSet<_>>();
-
-        let is_need_removed = used_idents.iter().any(|ident| match ident {
-          // need remove
-          UsedIdent::SwcIdent(_) => true,
-          // save all
-          UsedIdent::Default => true,
-          _ => false,
-        });
-
-        if !is_need_removed
-          || (!used_idents_string.is_empty() && used_idents.contains(&UsedIdent::Default))
-        {
-          continue;
-        }
-
-        for item_id in ids.iter().filter(|item_id| &item_id.index() == stmt) {
-          let item = items_map.get_mut(item_id).unwrap();
-
-          let mut removed_idents = item
-            .read_vars
-            .iter()
-            .enumerate()
-            .filter_map(|(index, id)| {
-              if !used_idents_string.contains(&Ident::from(id.clone()).to_string()) {
-                Some(index)
-              } else {
-                None
-              }
-            })
-            .collect::<Vec<_>>();
+    // 2. trace used statements starting from used exports
+    self
+      .stmt_graph
+      .trace_and_mark_used_statements(stmt_used_idents_map)
+  }
 
-          removed_idents.sort();
-          removed_idents.reverse();
+  fn all_exports_to_statement_idents(&self) -> Vec<(UsedStatementIdent, StatementId)> {
+    let mut used_idents = vec![];
 
-          for index in removed_idents {
-            item.read_vars.remove(index);
+    for export_info in self.exports() {
+      for sp in &export_info.specifiers {
+        match sp {
+          ExportSpecifierInfo::Default => {
+            used_idents.push((UsedStatementIdent::Default, export_info.stmt_id));
```

Comment:
`export * from` should not contain `default`

[MDN](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export)

> This re-exports all named exports from mod as the named exports of the current module, but the default export of mod is not re-exported.